src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -768,8 +770,9 @@ def get_result_set(
     def get_result_and_metadata(
         self, plan: SnowflakePlan, **kwargs
     ) -> Tuple[List[Row], List[Attribute]]:
-        result_set, result_meta = self.get_result_set(plan, **kwargs)
-        result = result_set_to_rows(result_set["data"])
+        show_limit = kwargs.pop("show_limit", None)
+        result_set, result_meta = self.get_result_set(plan, to_iter=True, **kwargs)
+        result = result_set_to_rows(result_set["data"], show_limit=show_limit)
         attributes = convert_result_meta_to_attribute(result_meta, self.max_string_size)
```

Comment:
how about naming the param `limit` to make it more general rather than `show_limit`.